File: src/main/java/me/testaccount666/serversystem/updates/DisabledUpdateChecker.java
package me.testaccount666.serversystem.updates;

import me.testaccount666.serversystem.utils.Version;

import java.util.concurrent.CompletableFuture;

public class DisabledUpdateChecker extends AbstractUpdateChecker {
    public DisabledUpdateChecker() {
        super(null);
        setAutoUpdate(false);
    }

    @Override
    public CompletableFuture<Boolean> hasUpdate() {
        return CompletableFuture.completedFuture(false);
    }

    @Override
    public CompletableFuture<Boolean> downloadUpdate() {
        return CompletableFuture.completedFuture(false);
    }

    @Override
    protected Version parseLatestVersion(String responseBody) {
        return null;
    }

    @Override
    protected String getDownloadUrl() {
        return "";
    }

    @Override
    public void setAutoUpdate(boolean autoUpdate) {
        super.setAutoUpdate(false);
    }
}

File: src/main/kotlin/me/testaccount666/serversystem/updates/DisabledUpdateChecker.kt
package me.testaccount666.serversystem.updates

import me.testaccount666.serversystem.utils.Version
import java.util.concurrent.CompletableFuture

class DisabledUpdateChecker : AbstractUpdateChecker(null) {

    override var autoUpdate: Boolean
        get() = false
        set(_) {}

    override fun hasUpdate(): CompletableFuture<Boolean> = CompletableFuture.completedFuture(false)

    override fun downloadUpdate(): CompletableFuture<Boolean> = CompletableFuture.completedFuture(false)

    override fun parseLatestVersion(responseBody: String): Version? = null

    override val downloadUrl: String
        get() = ""
}